synthesizer/synthesize.py

```diff
@@ -1,82 +1,97 @@
+    else:
+        device = torch.device("cpu")
+    print("Synthesizer using device:", device)
+
+    # Instantiate Tacotron model
+    model = Tacotron(embed_dims=hparams.tts_embed_dims,
+                     num_chars=len(symbols),
+                     encoder_dims=hparams.tts_encoder_dims,
+                     decoder_dims=hparams.tts_decoder_dims,
+                     n_mels=hparams.num_mels,
+                     fft_bins=hparams.num_mels,
+                     postnet_dims=hparams.tts_postnet_dims,
+                     encoder_K=hparams.tts_encoder_K,
+                     lstm_dims=hparams.tts_lstm_dims,
+                     postnet_K=hparams.tts_postnet_K,
+                     num_highways=hparams.tts_num_highways,
+                     dropout=0., # Use zero dropout for gta mels
+                     stop_threshold=hparams.tts_stop_threshold,
+                     speaker_embedding_size=hparams.speaker_embedding_size).to(device)
+
+    # Load the weights
+    model_dir = Path(model_dir)
+    model_fpath = model_dir.joinpath(model_dir.stem).with_suffix(".pt")
+    print("\nLoading weights at %s" % model_fpath)
+    model.load(model_fpath)
+    print("Tacotron weights loaded from step %d" % model.step)
+
+    # Synthesize using same reduction factor as the model is currently trained
+    r = np.int32(model.r)
+
+    # Set model to eval mode (disable gradient and zoneout)
+    model.eval()
+
+    # Initialize the dataset
+    in_dir = Path(in_dir)
+    metadata_fpath = in_dir.joinpath("train.txt")
+    mel_dir = in_dir.joinpath("mels")
+    embed_dir = in_dir.joinpath("embeds")
+
+    dataset = SynthesizerDataset(metadata_fpath, mel_dir, embed_dir, hparams)
+    data_loader = DataLoader(dataset,
+                             collate_fn=lambda batch: collate_synthesizer(batch, r),
+                             batch_size=hparams.synthesis_batch_size,
+                             num_workers=2,
+                             shuffle=False,
+                             pin_memory=True)
+
+    # Generate GTA mels
     meta_out_fpath = os.path.join(out_dir, "synthesized.txt")
     with open(meta_out_fpath, "w") as file:
-        for i, meta in enumerate(tqdm(metadata)):
-            texts = [m[5] for m in meta]
-            mel_filenames = [os.path.join(mel_dir, m[1]) for m in meta]
-            embed_filenames = [os.path.join(embed_dir, m[2]) for m in meta]
-            basenames = [os.path.basename(m).replace(".npy", "").replace("mel-", "") 
-                         for m in mel_filenames]
-            synth.synthesize(texts, basenames, synth_dir, None, mel_filenames, embed_filenames)
-            
-            for elems in meta:
-                file.write("|".join([str(x) for x in elems]) + "\n")
-                
-    print("Synthesized mel spectrograms at {}".format(synth_dir))
-    return meta_out_fpath
+        for i, (x, m, e, idx) in tqdm(enumerate(data_loader), total=len(data_loader)):
```

Comment:
Let's put slightly more informative names for x, m and e here

Reply:
Fixed in https://github.com/CorentinJ/Real-Time-Voice-Cloning/pull/472/commits/9187c25c20e7f31cbc86ae29aef1b0ec25e7cb94